build-gluster-org/scripts/generic_package.sh

```diff
@@ -0,0 +1,200 @@
+
+# to run use: 'bash -x generic_package.sh debian stretch 6 4.1.0 1'
+
+#Key generation: https://help.ubuntu.com/community/GnuPrivacyGuardHowto#Generating_an_OpenPGP_Key
+#Addditional info: sending of pubkey to a keyserv is been done using the MIT keyserv
+#################################################################################################
+os=$1
+flavor=$2
+series=$3
+version=$4
+release=$5
+latest_series=$6
+latest_version=$7
+
+#Keys required in debian builds
+declare -a debuild_keys
+debuild_keys=("8B7C364430B66F0B084C0B0C55339A4C6A7BD8D4",
+              "55F839E173AC06F364120D46FA86EEACB306CEE1",
+              "32F8E2FDBE1460F94A62407E468C889BEEDF12A8",
+              "F9C958A3AEE0D2184FAD1CBD43607F0DC2F8238C")
+
+declare -a pbuild_keys
+pbuild_keys=("7F6E5563", "EFCE7625", "4061252D", "BF11C87C")
+
+# Check for OS(Ubuntu or Debian)
+if [ "$os" == "ubuntu" ]; then
+        mirror="http://ubuntu.osuosl.org/ubuntu/"
+	debuild_key=4F5B5CA5
+elif [ "$os" == "debian" ]; then
+        mirror="http://ftp.us.debian.org/debian/"
+	case ${series} in
+	  "3.12")
+    	    debuild_key=${debuild_keys[0]}
+    	    pbuild_key=${pbuild_keys[0]}
+    	    ;;
+  	  "4.0")
+            debuild_key=${debuild_keys[1]}
+            pbuild_key=${pbuild_keys[1]}
+            ;;
+          "4.1")
+            if [ "$flavor" == "stretch" ]; then
+                 debuild_key=${pbuild_keys[2]}
+            else
+                 debuild_key=${debuild_keys[2]}
+            fi
+            pbuild_key=${pbuild_keys[2]}
+            ;;
+          "5" | "6" | "7")
+            if [ "$flavor" == "stretch" ]; then
+                 debuild_key=${pbuild_keys[3]}
+            else
+                 debuild_key=${debuild_keys[3]}
+            fi
+            pbuild_key=${pbuild_keys[3]}
+        esac
+else
+	echo "Exiting: OS should be debian or ubuntu. Please provide the right one"
+	exit
+fi
+
+mkdir ${os}-${flavor}-Glusterfs-${version}
+
+cd ${os}-${flavor}-Glusterfs-${version}
+
+mkdir build packages
+
+echo "Building glusterfs-${version}-${release} for ${flavor}"
+
+cd build
+
+TGZS=(`ls ~/glusterfs-${version}-?-*/build/glusterfs-${version}.tar.gz`)
+echo ${TGZS[0]}
+
+if [ -z ${TGZS[0]} ]; then
+        echo "wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz"
+        wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz
+else
+        echo "found ${TGZS[0]}, using it..."
+        cp ${TGZS[0]} .
+fi
+
+echo "Creating link file.."
+ln -s glusterfs-${version}.tar.gz glusterfs_${version}.orig.tar.gz
+
+echo "Untaring.."
+tar xpf glusterfs-${version}.tar.gz
+
+# Changelogs needed for building are maintained in a separate repo.
+# the repo has to be clone and updated properly so we can copy the changelogs so far.
+
+echo "Cloning the glusterfs-debian repo"
+git clone http://github.com/gluster/glusterfs-debian.git
```

Comment:
This should be done over ssh if you want to commit, and https anyway, to avoid MITM. I am not sure github support http.

Reply:
addressed